### PR TITLE
Remove secret generation of ironic CA cert in BMO.

### PR DESCRIPTION
**Pull request by smoshiur1237**
This PR will remove the secret generation of ironic CA certificate in BMO and use the same CA certificate secret name generated by cert-manager in ironic deployment. This PR is needed to able to run the [PR#859](https://github.com/metal3-io/baremetal-operator/pull/859) in BMO.

### CODE REVIEW

**Comment by fmuyassarov** [comment]
/retitle
🌱 WIP: Use cert-manager generated self-sigend ironic CA secret name for BMO.

**Comment by fmuyassarov** [comment]
/retitle 
 WIP: 🌱Use cert-manager generated self-sigend ironic CA secret name for BMO.

**Comment by kashifest** [comment]
Will this pass the integration test @smoshiur1237 ?

**Comment by kashifest** [comment]
LGTM, I would wait for others to review before I approve it.

**Comment by smoshiur1237** [comment]
> Will this pass the integration test @smoshiur1237 ?

It will not pass the integration test with the current ironic and bmo setup as we delete the option for secret creation. This PR is needed to run the BMO PR.

**Review by namnx228 — APPROVED** [review]
LGTM

**Review by kashifest — APPROVED** [review]

**Comment by levsha** [comment]
This PR broke the scenario when an ironic deployment without SSL is used and IRONIC_NO_CA_CERT is set.

**Comment by fmuyassarov** [comment]
> This PR broke the scenario when an ironic deployment without SSL is used and IRONIC_NO_CA_CERT is set.

/cc @smoshiur1237

**Comment by smoshiur1237** [comment]
> This PR broke the scenario when an ironic deployment without SSL is used and IRONIC_NO_CA_CERT is set.

Working on the fix and PR opened to fix the issue with non-TLS.  [CAPM3](https://github.com/metal3-io/cluster-api-provider-metal3/pull/237), [BMO](https://github.com/metal3-io/baremetal-operator/pull/924) and [metal3-dev](https://github.com/metal3-io/metal3-dev-env/pull/720) PR

**Pull request merged**
